masquerade-maven-plugin/src/main/java/com/flipkart/masquerade/Masque.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2017 Flipkart Internet, pvt ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.flipkart.masquerade;
+
+import org.apache.maven.artifact.DependencyResolutionRequiredException;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugin.logging.Log;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.plugins.annotations.ResolutionScope;
+import org.apache.maven.project.MavenProject;
+
+import java.io.File;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by shrey.garg on 12/05/17.
+ */
+@Mojo(
+        name = "masquerade",
+        requiresDependencyCollection = ResolutionScope.COMPILE_PLUS_RUNTIME,
+        requiresDependencyResolution = ResolutionScope.COMPILE_PLUS_RUNTIME
+)
+public class Masque extends AbstractMojo {
+
+    @Parameter(required = true)
+    private String configurationClass;
+
+    @Parameter(required = true)
+    private File targetFile;
```

Comment:
Do we need to take this configuration or make assumption around the placement of generated files? If someone gives this as src folder (as done in sample), generated java files are placed in package "org.test.veils" under masquerade-sample/src/main/java/.
Now mvn clean won't clean this folder hence older files based on old config could lie around forever.

Ideally they should be always under target/generated-sources (at least for mvn projects)